android/src/com/frostwire/android/core/server/CastFileServer.java

```diff
@@ -0,0 +1,216 @@
+                    int minus = range.indexOf('-');
+                    try {
+                        if (minus > 0) {
+                            startFrom = Long.parseLong(range
+                                    .substring(0, minus));
+                            endAt = Long.parseLong(range.substring(minus + 1));
+                        }
+                    } catch (NumberFormatException ignored) {
+                    }
+                }
+            }
+
+            long fileLen = file.length();
+            if (range != null && startFrom >= 0) {
+                if (startFrom >= fileLen) {
+                    response = newFixedLengthResponse(Response.Status.RANGE_NOT_SATISFIABLE,
+                            NanoHTTPD.MIME_PLAINTEXT, "");
+                    response.addHeader(ACCEPT_RANGES_HEADER, "bytes");
+                    response.addHeader(CONTENT_RANGE_HEADER, "bytes 0-0/" + fileLen);
+                    response.addHeader(E_TAG_HEADER, eTag);
+                } else {
+                    if (endAt < 0) {
+                        endAt = fileLen - 1;
+                    }
+                    long newLen = endAt - startFrom + 1;
+                    if (newLen < 0) {
+                        newLen = 0;
+                    }
+
+                    final long dataLen = newLen;
+                    FileInputStream fis = new FileInputStream(file) {
+                        @Override
+                        public int available() throws IOException {
+                            return (int) dataLen;
+                        }
+                    };
+                    long skipped = fis.skip(startFrom);
+                    if (skipped != startFrom) {
+                        LOG.warn("Problem skipping, wanted: " + startFrom + " got: " + skipped);
+                    }
+                    response = newChunkedResponse(Response.Status.PARTIAL_CONTENT, mime,
+                            fis);
+                    response.addHeader(ACCEPT_RANGES_HEADER, "bytes");
+                    response.addHeader(CONTENT_LENGTH_HEADER, "" + dataLen);
+                    response.addHeader(CONTENT_RANGE_HEADER, "bytes " + startFrom + "-"
+                            + endAt + "/" + fileLen);
+                    response.addHeader(E_TAG_HEADER, eTag);
+                }
+            } else {
+                if (eTag.equals(header.get("if-none-match"))) {
+                    response = newFixedLengthResponse(Response.Status.NOT_MODIFIED, mime, "");
+                    response.addHeader(ACCEPT_RANGES_HEADER, "bytes");
+                } else {
+                    response = newChunkedResponse(Response.Status.OK, mime,
+                            new FileInputStream(file));
+                    response.addHeader(ACCEPT_RANGES_HEADER, "bytes");
+                    response.addHeader(CONTENT_LENGTH_HEADER, "" + fileLen);
+                    response.addHeader(E_TAG_HEADER, eTag);
+                }
+            }
```

Comment:
I believe a call to 

```java
try {
    response.close()
} catch (Throwable t) { //close quietly }
``` 

is missing here, otherwise the passed `FileInputStream` remains open... unless whoever calls `serveFile` closes it, but I don't see that in the file anywhere.